BootloaderCommonPkg/Library/SecureBootLib/SecureBootHash.c

```diff
@@ -13,6 +13,54 @@
 #include <Library/SecureBootLib.h>
 #include <Library/BootloaderCommonLib.h>
 
+
+/**
+  Calculate hash API.
+
+  @param[in]  Data           Data buffer pointer.
+  @param[in]  Length         Data buffer size.
+  @param[in]  HashAlg        Specify hash algrothsm.
+  @param[in,out]  OutHash       On input,  expected hash value when ComponentType is not used.
+                             On output, calculated hash value.
+
+  @retval RETURN_SUCCESS             Hash verification succeeded.
+  @retval RETRUN_INVALID_PARAMETER   Hash parameter is not valid.
+  @retval RETURN_UNSUPPORTED         Hash component type is not supported.
+
+**/
+
+RETURN_STATUS
+DoHashCalc (
+  IN CONST UINT8          *Data,
+  IN       UINT32          Length,
+  IN       UINT8           HashAlg,
+  IN OUT   UINT8          *OutHash
+  )
+{
+  UINT8 Digest[HASH_DIGEST_MAX];
+  UINT8 DigestSize;
+
+  if(Digest != NULL){
+    if (HashAlg == HASH_TYPE_SHA256) {

```

Comment:
Should you check your build PCD too ?
Otherwise, all algorithm will be built in.

Reply:
PcdIppHashLibSupportedMask is controlled in wrapper libraries as Sha256.c.
Incase respective mask is not enabled, hash calc would not happen. Need to add error checks where ever we use when Hash functions were invoked.

Reply:
Ok,. please check image size again after your change.  It should change for current projects.

Reply:
I still think this might have impact on size.
If a project does not use SM3 and only use SHA256, 
how will the compiler know that the line #49 and #46 should be removed ?

Reply:
Better to collect some data to see if it has impact to size.

Reply:
You checked the PCD inside the internal function.  Got it. 
That should be fine then.

Reply:
Thanks for checking!